src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -280,8 +281,17 @@ public ResultWrapper<byte[]> eth_sign(Address addressData, byte[] message)
 
     public virtual Task<ResultWrapper<Hash256>> eth_sendTransaction(TransactionForRpc rpcTx)
     {
-        Transaction tx = rpcTx.ToTransactionWithDefaults(_blockchainBridge.GetChainId());
-        TxHandlingOptions options = rpcTx.Nonce is null ? TxHandlingOptions.ManagedNonce : TxHandlingOptions.None;
+        Transaction tx = rpcTx.ToTransaction();
+        tx.ChainId = _blockchainBridge.GetChainId();
+
+        // TODO: We might want to lift `Nonce` to `TransactionForRpc`
+        UInt256? nonce = null;
+        if (rpcTx is LegacyTransactionForRpc legacyTx)
+        {
+            nonce = legacyTx.Nonce;
+        }
```

Comment:
As the comment says, should we lift `Nonce` to the base `TransactionForRpc` class?

Reply:
Yes, no idea why it isn't there in the first place

Reply:
It was not on the base type since new Transaction types might not have a `Nonce`. For example, in Optimism we kind of don't have a Nonce (see https://github.com/ethereum-optimism/op-geth/blob/8af19cf20261c0b62f98cc27da3a268f542822ee/core/types/deposit_tx.go#L29-L46)

Reply:
Will deposit transaction interact with JSON RPC? Shouldn't it nonce be expected to be outputted there?

Otherwise: true, but `Nonce` sounds like such a basic transaction functionality that maybe it is better to make an exception, put it in base class and just ignore it or `DepositTransaction`? Or this shouldn't take in `TransactionForRpc` but `LegacyTransactionForRpc`? Or maybe create a class in between called like `UserTransactionForRpc`?